services/ui-src/src/components/overlays/EntityDetailsMultiformOverlay.tsx

```diff
@@ -132,23 +138,27 @@ export const EntityDetailsMultiformOverlay = ({
         const updatedData = { ...enteredData };
 
         if (isCompliant) {
-          // Set all standard keys to undefined
+          // delete all standard keys
           allStandardKeys.forEach((key) => {
-            updatedData[key] = undefined;
+            delete selectedEntity?.[key];
+            delete updatedData[key];
```

Comment:
again, need to delete keys from `selectedEntity` as well in case the user changed their choice